### PR TITLE
Enable periodic cleanup of expired outbox records

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Wires up the outbox cleaner task using the defaults same as in NHibernate persister:
 * Cleanup interval 1 minute
 * Outbox record life 7 days

### CODE REVIEW

**Comment by SimonCropp** [comment]
@SzymonPobiega i did some testing and found a few more issues. so i ported the test from develop to verify. can u QA please?

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs [hunk 1]
```diff
@@ -148,4 +149,63 @@ public void StoreAndGet()
         return await persister.Get(messageId, null);
     }
 
+    [Test]
+    public async Task StoreAndCleanup()
+    {
+        using (var connection = await dbConnection.OpenConnection())
+        {
+            for (var i = 0; i < 13; i++)
```
Comment (SzymonPobiega) [inline src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs]: Why 13? Wouldn't 1 be enough?

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs [hunk 2]
```diff
@@ -148,4 +149,63 @@ public void StoreAndGet()
...
+            {
+                await Store(i, connection);
+            }
+        }
+
+        await Task.Delay(1000);
```
Comment (SzymonPobiega) [inline src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs]: What do you think about passing `Func<DateTime>` to `OutboxPersister` in `ctror` to make this tests not time-dependent?

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs (hunk 1 above)
Comment (SimonCropp) [inline src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs]: well i wanted more than 1 so multiple would be cleaned up. so any number greater than 1 would do

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs (hunk 2 above)
Comment (SimonCropp) [inline src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs]: happy to. but i think that can be an optimisation on develop?

**Pull request merged**